src/test/java/seedu/ddd/model/contact/common/predicate/ContactPredicateBuilderTest.java

```diff
@@ -0,0 +1,266 @@
+package seedu.ddd.model.contact.common.predicate;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.ddd.logic.parser.CliFlags.FLAG_CLIENT;
+import static seedu.ddd.logic.parser.CliFlags.FLAG_VENDOR;
+import static seedu.ddd.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static seedu.ddd.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.ddd.logic.parser.CliSyntax.PREFIX_ID;
+import static seedu.ddd.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.ddd.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.ddd.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.ddd.testutil.TypicalContacts.ALICE;
+import static seedu.ddd.testutil.TypicalContacts.BENSON;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.ddd.logic.parser.ArgumentMultimap;
+import seedu.ddd.logic.parser.exceptions.ParseException;
+import seedu.ddd.testutil.ClientBuilder;
+
+public class ContactPredicateBuilderTest {
+
+    @Test
+    public void equals() {
+        ArgumentMultimap argMultimap1 = new ArgumentMultimap();
+        argMultimap1.put(PREFIX_NAME, "Alice Millow");
+        ContactPredicateBuilder firstPredicate = new ContactPredicateBuilder(argMultimap1);
+
+        ArgumentMultimap argMultimap2 = new ArgumentMultimap();
+        argMultimap2.put(PREFIX_ID, "1");
+        ContactPredicateBuilder secondPredicate = new ContactPredicateBuilder(argMultimap2);
+
+        // same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        ContactPredicateBuilder firstPredicateCopy =
+                new ContactPredicateBuilder(argMultimap1);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different name list -> returns false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
+
+    @Test
+    public void testArgumentMultimapContainsNameReturnsTrue() throws ParseException {
+        ArgumentMultimap argMultimap = new ArgumentMultimap();
+        argMultimap.put(PREFIX_NAME, "Alice Millow");
+        ContactPredicateBuilder predicateBuilder = new ContactPredicateBuilder(argMultimap);
+
+        assertTrue(predicateBuilder.build().test(new ClientBuilder().withName("Alice Millow").build()));
+        assertTrue(predicateBuilder.build().test(new ClientBuilder().withName("Alice").build()));
+        assertTrue(predicateBuilder.build().test(new ClientBuilder().withName("milLow").build()));
+    }
+
+    @Test
+    public void testArgumentMultimapDoesNotContainNameReturnsFalse() throws ParseException {
```

Comment:
Nitpick but the recommended test name format is

```
featureUnderTest_testScenario_expectedBehavior()
```

So maybe can change this to `testArgumentMultiMap_containsName_returnsTrue()` instead.

Reply:
This applies to the other tests too.